target/arc/op_helper.c

```diff
@@ -423,37 +423,6 @@ target_ulong helper_carry_add_flag(target_ulong dest, target_ulong b,
     return carry_add_flag(dest, b, c, TARGET_LONG_BITS);
 }
 
-static inline target_ulong
-overflow_add_flag(target_ulong dest, target_ulong b, target_ulong c,
-                  uint8_t size)
-{
-    dest >>= (size - 1);
-    b    >>= (size - 1);
-    c    >>= (size - 1);
-
-    /*
-     * Truncate the result.
-     * Negative numbers may have 1s to the left of the 'size'th bit when the
-     * architecture has more bits than the requested operation i.e. 32 bit
-     * operation in a 64 bit architecture has the most significant 32 bits at 1
-     * for any and all negative numbers.
-     */
-    dest &= 1;
```

Comment:
isn't `tcg_gen_shri_tl()` at the core of `arc_gen_add_signed_overflow_tl()` susceptible to the same problem? Even if not, I think a `tcg_gen_andi(..., 1)` is in order, to return only and only `1` or `0` in `overflow`.

I'm thinking of a 32-bit signed addition inside a 64-bit TCGv:
```
  |-SGN-|
0b111...1xxx...x op1
0b111...1xxx...x op2
0b000...0xxx...x res
--------------------
0b000...0xxx...x diff_operand_sign (op1 ^ op2)
0b111...1xxx...x diff_result_sign  (op1 ^ res)
0b111...1xxx...x overflow          (before shri 32)
0b000...0111...1 overflow          (after shri 32)
```

Reply:
It's not susceptible to the same problem, because if the "user" of the function matches against 0, it is correct (the return is the difference in signs).
But you make a valid point, the assumption above contradicts the comment, so its better to add and andi(..., 1) to make the function return 0/1 instead of0/non-zero, so we avoid confusion in the future.